src/parser/Lexer.cpp

```diff
@@ -1941,7 +1917,18 @@ void Scanner::lex(Scanner::ScannerResult* token)
         return;
     }
 
-    const char16_t cp = this->peekCharWithoutEOF();
+    char16_t cp = this->peekCharWithoutEOF();
+
+    if (UNLIKELY(this->peekChar() >= 0xD800 && this->peekChar() < 0xDFFF)) {
+        cp = peekChar();
+        ++this->index;
+        char32_t ch2 = this->peekChar();
+        if (U16_IS_TRAIL(ch2)) {
+            cp = U16_GET_SUPPLEMENTARY(cp, ch2);
+        } else {
+            this->throwUnexpectedToken();
+        }
+    }
```

Comment:
It seems that this check code is the main reason of the overall performance slow down.
(`lex` is one of the most frequently called functions so I think that adding just one check code may affect the overall performance)
IMHO this code can be removed since unicode table is newly updated.
What do you think about it?

Reply:
Atm this code is needed for the characters that are in the Supplementary Range. For ex.:
```js
var 𑴀 = "foo"
print(𑴀)
```
will fail with ILLEGAL TOKEN error.

Reply:
One possible idea is to move this to the `isIdentifierPart` and `isIdentifierStart` part afther the :
```
if (LIKELY(ch < 128)) {
        return g_asciiRangeCharMap[ch] & LexerIsCharIdent;
    }
```
since at that part we are at "slow" mode anyway.

Reply:
Your idea looks great!
Would you please apply it and measure the performance once again?

Reply:
@clover2123 Decided to create some new "fast paths", by checking the ascii table and letting it through right on the spot without all the peeks. New Results:
https://gist.github.com/Achie72/dd0ec5e4f7fb83333086e246b241603d
It varies around this 605-610 area.